ONSdigital/dp-search-data-finder
# Update total search documents

# What has changed




The handler for the reindex-requested events has been updated so that after it has logged out the first ten URLs retrieved from zebedee it will then do the following:
- use the Search Reindex client to call the PATCH /jobs/{jobID} endpoint (using the jobID value from the consumed event) to update the job 'state' value to "in-progress" and its 'total_search_documents' value to be the total number of URLs retrieved from zebedee.
NB. The PATCH endpoint should then automatically set the 'reindex_started' value to be the current time and should return an ETag in its response header. 
- log the ETag that was returned (via the Search Reindex client) by the PATCH endpoint.

The unit test TestReindexRequestedHandler_Handle has been updated to reflect the changes in the handler.

The README has been updated to include the SEARCH_REINDEX_URL, SERVICE_AUTH_TOKEN, and ZEBEDEE_CLIENT_TIMEOUT environment variables.

# How to review

Check that the code looks correct and that the following tests pass:

make test

go test -component

make lint

To see the patch endpoint being called and the returned ETag being logged locally do as follows:
- in dp-compose run ElasticSearch and Kafka as follows: docker-compose up -d
- run zebedee
- run the Search API (but first run this command to use new ElasticSearch): export ELASTIC_SEARCH_URL="http://localhost:11200"
- run the Search Reindex API
- call the POST /jobs endpoint on the Search Reindex API to create a reindex-requested event:
curl -s http://localhost:25700/jobs | jq

NB. The job returned should look something like this:
{
    "ETag": "\"66317dc874abd7b4b9f7fe3dd44909e54a190207\"",
    "id": "4efe5b5b-3cef-4033-9729-50cd973b4e16",
    "last_updated": "2022-04-14T09:59:07.052681Z",
    "links": {
        "tasks": "http://localhost:25700/jobs/4efe5b5b-3cef-4033-9729-50cd973b4e16/tasks",
        "self": "http://localhost:25700/jobs/4efe5b5b-3cef-4033-9729-50cd973b4e16"
    },
    "number_of_tasks": 0,
    "reindex_completed": "0001-01-01T00:00:00Z",
    "reindex_failed": "0001-01-01T00:00:00Z",
    "reindex_started": "0001-01-01T00:00:00Z",
    "search_index_name": "ons1649930347064342",
    "state": "created",
    "total_search_documents": 0,
    "total_inserted_search_documents": 0
}

- run the dp-search-data-finder code to call the patch endpoint 
- call the GET /jobs/{id} endpoint to see the updated state, total_search_documents, and reindex_started time
e.g. for the job id above you would call http://localhost:25700/jobs/4efe5b5b-3cef-4033-9729-50cd973b4e16

and you would expect to see something like this:    

{
  "ETag": "\"0f36ac5a0d5a38aee682deac22a167647b5b604f\"",
  "id": "4efe5b5b-3cef-4033-9729-50cd973b4e16",
  "last_updated": "2022-04-26T10:22:26.558Z",
  "links": {
      "tasks": "localhost:25700/v1/jobs/4efe5b5b-3cef-4033-9729-50cd973b4e16/tasks",
      "self": "localhost:25700/v1/jobs/4efe5b5b-3cef-4033-9729-50cd973b4e16"
  },
  "number_of_tasks": 0,
  "reindex_completed": "0001-01-01T00:00:00Z",
  "reindex_failed": "0001-01-01T00:00:00Z",
  "reindex_started": "2022-04-26T10:22:26.558Z",
  "search_index_name": "ons1650968517386170",
  "state": "in-progress",
  "total_search_documents": 209,
  "total_inserted_search_documents": 0
}

# Who can review

Anyone but me
